[export] Load side info about pos/kw argument kind for serialization.

zhxchen17 opened the pull request:
Summary:
Fixing issue of nodes like
```
torch.ops.aten.linear.default(x, w, b)
```
being deserialized as
```
torch.ops.aten.linear.default(x, w, bias=b)
```
which breaks roundtripping.

Test Plan:
buck test mode/opt caffe2/test:test_export -- -r TestDeserialize
buck test mode/opt caffe2/test:test_export -- -r TestSerialize

Differential Revision: D67991410

avikchaudhuri commented:
@zhxchen17 Could you update `test_nn_module_stack_shared_submodule` using `export` instead of `torch.export.export`? IIRC it was failing because of this issue.

Internal task [T205027593](https://www.internalfb.com/intern/tasks/?t=205027593)

zhxchen17 committed:
[export] Load side info about pos/kw argument kind for serialization. (#144686)

Summary:

Fixing issue of nodes like
```
torch.ops.aten.linear.default(x, w, b)
```
being deserialized as
```
torch.ops.aten.linear.default(x, w, bias=b)
```
which breaks roundtripping.

Test Plan:
buck test mode/opt caffe2/test:test_export -- -r TestDeserialize
buck test mode/opt caffe2/test:test_export -- -r TestSerialize

Reviewed By: yiming0416

Differential Revision: D67991410
